provisioning/baremetal_secrets.go

```diff
@@ -118,29 +122,62 @@ password = %s
 // createRegistryPullSecret creates a copy of the pull-secret in the
 // openshift-config namespace for use with LocalObjectReference
 func createRegistryPullSecret(info *ProvisioningInfo) error {
-	secretClient := info.Client.CoreV1().Secrets(openshiftConfigNamespace)
-	openshiftConfigSecret, err := secretClient.Get(context.TODO(), pullSecretName, metav1.GetOptions{})
+	client := info.Client.CoreV1()
+	openshiftConfigSecret, err := client.Secrets(OpenshiftConfigNamespace).Get(context.TODO(), PullSecretName, metav1.GetOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("could not get secret %s/%s, err: %w", OpenshiftConfigNamespace, PullSecretName, err)
+	}
+	openshiftConfigSecretKeyData, ok := openshiftConfigSecret.Data[openshiftConfigSecretKey]
+	if !ok {
+		return fmt.Errorf("could not find key %q in secret %s/%s", openshiftConfigSecretKey, OpenshiftConfigNamespace, PullSecretName)
+	}
+
+	// Try to get the openshift-machine-api/pull-secret field .dockerconfigjson.
+	// The openshift-machine-api/pull-secret .dockerconfigjson field should be double encoded due to PR
```

Comment:
What do we mean by *double* encoded?

Reply:
In OCP, all secret data is base64 encoded, e.g.:
```
echo "string" | base64 -w0
```

That happens automatically when you write a string into Secret.StringData:
https://github.com/kubernetes/api/blob/v0.28.1/core/v1/types.go#L6541C3-L6541C3

It's then automatically base64 encoded and can be retrieved in Secret.Data:
https://github.com/kubernetes/api/blob/v0.28.1/core/v1/types.go#L6533

But the content of openshift-machine-api/pull-secret is actually 2x base64 encoded, the bash equivalent is:
```
echo "string" | base64 -w0 | base64 -w0
```

See https://github.com/openshift/cluster-baremetal-operator/pull/184/commits/067b314eebfc48befe4ad684fa1e9da1f91050d3#diff-fe5186bad7cb27f50e9cfd2a5ef8d36067e64477380926b9d6da32eeaa9f08ffR157

Unfortunately, that's existing logic that I didn't want to touch. It would IMO have been way better to store the secret correctly encoded ( = only once base64 encoded) and instead make sure that whatever component is responsible for sending the secret to the Configure-CoreOS init container (or the container itself) takes care of encoding this once more, if needed